cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -327,6 +334,11 @@ public ApplicationWithPrograms updateApp(ApplicationId appId, AppRequest appRequ
       newArtifactId = new ArtifactId(currentArtifact.getName(), requestedVersion, currentArtifact.getScope());
     }
 
+    // ownerAdmin.getOwner will give the owner of application irrespective of the version
+    boolean equals = Objects.equals(ownerAdmin.getOwner(appId), appRequest.getOwnerPrincipal());
```

Comment:
No there is no specific reason but I have some test at below Handler level AppLifeCycleServiceTest which test the deployment logic so added it here as another safety. Should we get rid of it ?